core/tests/protractor_desktop/coreEditorAndPlayerFeatures.js

```diff
@@ -0,0 +1,248 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for the core features of the exploration
+ * editor and player. Core features include the features without which an
+ * exploration cannot be published. These include state content, answer groups,
+ * oppia's feedback and customization_args.
+ */
+
+var forms = require('../protractor_utils/forms.js');
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var CreatorDashboardPage =
+  require('../protractor_utils/CreatorDashboardPage.js');
+var ExplorationEditorPage =
+  require('../protractor_utils/ExplorationEditorPage.js');
+var ExplorationPlayerPage =
+  require('../protractor_utils/ExplorationPlayerPage.js');
+var LibraryPage = require('../protractor_utils/LibraryPage.js');
+
+
+describe('Core exploration functionality', function() {
+  var explorationPlayerPage = null;
+  var explorationEditorPage = null;
+  var explorationEditorMainTab = null;
+
+  beforeEach(function() {
+    explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+    explorationEditorMainTab = explorationEditorPage.getMainTab();
+    explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+  });
+
+  it('should display plain text content', function() {
```

Comment:
How about we put all common setup steps in a `beforeAll` block? We can then create a single user for this complete `describe` block once and log it in. Then login (except for the first test) and logout the user for each test, some thing similar to [this](https://github.com/oppia/oppia/blob/develop/core/tests/protractor/profileMenuFlow.js#L29). There can be other common setup steps too. Would love to hear your thoughts on this!

Reply:
I didn't want to actually edit the tests as I wanted to make sure that the coverage is unaffected by the refactor. Happy to do this analysis and group stuff, if that's ok with you (might get slightly harder to review on the github UI). Will follow up a commit with a setup refactor.

Reply:
Will leave this analysis and optimization to happen during GSoC